### PR TITLE
 feat(ci): add rust check and codegen 

### PR DESCRIPTION
**Pull request by rgallor**
Add GitHub workflows:
- generate rust files from proto
- check rust code
- commit generated files
- dependabot for cargo dependencies

### CODE REVIEW

File: rust/edgehog-device-forwarder-proto/src/proto.rs
Comment (harlem88) [inline rust/edgehog-device-forwarder-proto/src/proto.rs:1]: Is it correct that this file has already been added? It should be generated after the pull_request...

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: rust/edgehog-device-forwarder-proto/src/proto.rs
Comment (rgallor) [reply rust/edgehog-device-forwarder-proto/src/proto.rs:1]: The file was generated manually in the commit before the CI one. It's the same as the one that would be generated by the CI and would be overwritten. If you want I can still remove it.

File: .github/workflows/commit-codegen.yaml
Comment (harlem88) [inline .github/workflows/commit-codegen.yaml:29]: We are considering whether it is better to create a separate PR instead of doing direct push to the master branch... see here https://github.com/astarte-platform/astarte-message-hub-proto/pull/3/commits/8b44fcb6328d56a9ab681c5cab8a743bbdac6ff8#diff-47c440ec0726ad4c1976e840922373c6de7839fd55e82210c1b0e1eb93ce8d71

Comment (rgallor) [reply .github/workflows/commit-codegen.yaml:29]: I think it could be a good idea. I can implement the same thing

File: scripts/codegen.sh
Comment (harlem88) [inline scripts/codegen.sh:8]: I think we can skip the check on `protoc_version`, it should be the `protoc_version` used by the committer to generate the code, but this is not possible.

File: Makefile
Comment (harlem88) [inline Makefile:1]: Do we really need this file?

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: .github/workflows/codegen-check.yaml
Comment (harlem88) [inline .github/workflows/codegen-check.yaml:15]: Bump to the latest version
```suggestion
  PB_VERSION: 24.4
```

File: scripts/codegen.sh
Comment (harlem88) [inline scripts/codegen.sh:8]:
```suggestion
PROTOC_VERSION='24.4'
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: scripts/codegen.sh
Comment (rgallor) [reply scripts/codegen.sh]: Done

File: .github/workflows/codegen-check.yaml
Comment (rgallor) [reply .github/workflows/codegen-check.yaml]: Done

**Review by harlem88 — APPROVED** [review]
This PR depends on https://github.com/edgehog-device-manager/edgehog-device-forwarder-proto/pull/1, so it should be merged after it.

**Review by harlem88 — APPROVED** [review]

**Review by noaccOS — CHANGES_REQUESTED** [review]

File: Makefile
Comment (joshuachp) [reply Makefile:59]: Is there a reason to use a different version?

The check is necessary to ensure that the established version of Protoc is used while committing, it doesn't make sense to permit the use of a different version here.

Warnings wouldn't help, since they can be ignored.

File: scripts/rust_codegen.sh
Comment (joshuachp) [inline scripts/rust_codegen.sh:25]: Install by default give executable permission to the files

**Review by joshuachp — CHANGES_REQUESTED** [review]

File: scripts/rust_codegen.sh
Comment (joshuachp) [reply scripts/rust_codegen.sh:25]:
```suggestion
    install -m 644 "$OUT_DIR"/edgehog.device.forwarder.rs "$INSTALL_DIR"/edgehog-device-forwarder-proto/src/proto.rs
```

File: Makefile
Comment (noaccOS) [reply Makefile:59]: there are 4 possible outcomes when compiling with a different protoc version
1. it compiles the code as expected
2. it crashes, in which case the `make` command fails
3. it compiles successfully but the code it generates is not the same as with `$PROTOC_VERSION`

for 2., the warning will point to the possible cause of the failure
for 3., the CI check will fail anyway

I don't think it's necessary to force the user to use the version we want, but we should inform them of the _officially supported_ version

Comment (joshuachp) [reply Makefile:59]: This script is _for us and nobody will ever run it_. It's more beneficial to make the build brake on an incorrect version than warn about it.

Even if somebody runs this they will be aware of the wrong version and download the appropriate version or comment out the check.

> Is there a reason to use a different version?

Also, to drive the point home, the build should break if the proto version is wrong. Since there is no reason that _we_ would want to commit files generated with a different version of `protoc`.

Maybe a compromise would be to have an environment variable to pass this check, but I believe there should be an actual reason to use a different version of protoc.

Comment (joshuachp) [reply Makefile:59]: This already happened when we updated from proto 24.3 → 24.4, having the protoc check prevents the user from uploading the wrong code and having to wait for CI to fail with just the diffs being incompitible.

**Review by noaccOS — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

File: scripts/rust_codegen.sh
Comment (rgallor) [reply scripts/rust_codegen.sh]: Done

**Pull request merged**
